packages/salesforce/test/Adaptor.test.js

```diff
@@ -194,4 +195,124 @@ describe('Adaptor', () => {
       expect(result).to.eql('Nhamaonha 6a Classe 2023-10-09');
     });
   });
+
+  describe('query', () => {
+    it('should properly pass the query string', async () => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+
+      let state = {
+        connection: fakeConnection,
+        references: [],
+        qs: 'select Name from Account',
+      };
+      let spy = sinon.spy(fakeConnection, 'query');
+
+      query('select Name from Account')(state).then(state => {
+        expect(spy.calledWith('select Name from Account')).to.eql(true);
+      });
+
+      query(state => state.qs)(state).then(state => {
```

Comment:
Why is this duplicated?

if you're testing that references can be expanded, I think that's a different test entirely

A unit test should test one thing and one thing only. Every character of code in a test should support that one case (either setting it up or asserting it worked). Every unrelated character of code weakens the test.

Reply:
I have separated the tests